internal/server/repository_worker.go

```diff
@@ -198,27 +204,30 @@ func ListWorkers(ctx context.Context, reader db.Reader, scopeIds []string, opt .
 		where = append(where, fmt.Sprintf("last_status_time > now() - interval '%d seconds'", uint32(liveness.Seconds())))
 	}
 	if len(scopeIds) > 0 {
-		where = append(where, "scope_id in (?)")
-		whereArgs = append(whereArgs, scopeIds)
+		where = append(where, "w.scope_id in (@scope_id)")
+		whereArgs = append(whereArgs, sql.Named("scope_id", strings.Join(scopeIds, ",")))
 	}
 
+	// Note: this option will be removed in a future PR for this llb
 	switch opts.withWorkerType {
 	case "":
 	case KmsWorkerType, PkiWorkerType:
-		where = append(where, "type = ?")
-		whereArgs = append(whereArgs, opts.withWorkerType.String())
+		where = append(where, "w.type = @type")
+		whereArgs = append(whereArgs, sql.Named("type", opts.withWorkerType.String()))
 	default:
 		return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("unknown worker type %v", opts.withWorkerType))
 	}
 
+	// Note: this option will be removed in a future PR for this llb
```

Comment:
Is there a reason to wait for a future PR? It seems like this is making changes that will just be dropped soon which just leads to more changes to review that might not even remain.

Reply:
We have incoming work to create domain functions that replace usages of ListWorkers where these options are passed; we can't remove these options until those are created, and some of the refactoring to make those domain functions will be significant.